browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -98,19 +100,35 @@ void RewardsTabHelper::DidFinishLoad(
   rewards_service_->OnLoad(tab_id_, validated_url);
 }
 
-void RewardsTabHelper::DidFinishNavigation(content::NavigationHandle* handle) {
-  if (!handle->IsInMainFrame() || !handle->HasCommitted() ||
-      handle->IsDownload()) {
+void RewardsTabHelper::DidFinishNavigation(
+    content::NavigationHandle* navigation_handle) {
+  if (!navigation_handle->HasCommitted() ||
+      !navigation_handle->IsInPrimaryMainFrame() ||
+      navigation_handle->IsDownload()) {
     return;
   }
 
-  auto id = GetPublisherIdFromURL(GetWebContents().GetLastCommittedURL());
-  SetPublisherIdForTab(id ? *id : "");
+  if (!rewards_service_) {
+    return;
+  }
 
-  MaybeSavePublisherInfo();
+  auto& url = GetWebContents().GetLastCommittedURL();
```

Comment:
don't use `GetWebContents().GetLastCommittedURL()` when NavigationHandle is available

Reply:
Fixed - thanks.